core/DataAccess/ArchiveWriter.php

```diff
@@ -197,6 +197,15 @@ public function initNewArchive()
 
     public function finalizeArchive()
     {
+        if (
+            empty($this->recordsToWriteSpool['blob'])
+            && count($this->recordsToWriteSpool['numeric']) === 1
+            && $this->recordsToWriteSpool['numeric'][0][0] === $this->doneFlag
+            && $this->parameters->isPartialArchive()
```

Comment:
Is it possible to make this clearer what is actually being checked?

Is this all the checks needed to determine if partial archive?